src/hooks/NoInternetToastProvider.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { createContext, useEffect, useRef, useState } from 'react';
+import { useNetInfo } from '@react-native-community/netinfo';
+import { t } from 'i18next';
+import Toast, { ToastProps } from 'react-native-toast-message';
+
+export const NoInternetToastContext = createContext({});
+
+export const NoInternetToastProvider = ({
+  children,
+  toastOverrides,
+  forceShowToast,
+}: {
+  children?: React.ReactNode;
+  toastOverrides?: ToastProps;
+  forceShowToast?: boolean;
```

Comment:
Yet another nit-pick, sorry I have so many on such a simple component: I don't love how we're exposing a property that's solely for testing purposes.  I know that's done sometimes, perhaps with `_` prefix, but I think it should be avoided whenever possible.  In this case, I wonder about just making the story show some text like "Turn off internet on your machine, or in your simulator, to see the No Internet toast".  It doesn't make for as smooth/intuitive of a story/demo as what you have, but between the story and not exposing extra props, I'd vote for the latter.  Alternatively, if we do take in a testing prop, I think just taking an injected `isConnected` override might be simpler.  I might be reading it wrong (lmk) but I feel like the `forceShowToast` checks get confusing in the useEffect.

Reply:
I've been digging into alternatives but coming up empty. Unfortunately the connection state change events from `useNetInfo` do not work properly on an iOS simulator so we cannot rely on expecting folks to disable their connection to see this component. As far as I can tell our only options are to:

A) Remove the storybook example
B) Keep the extra prop but rename it and add JSDoc to explain it better, something like this:

![image](https://user-images.githubusercontent.com/76954025/222280993-0b7c415f-0998-4305-b6a5-73268568af06.png)

Reply:
If we go for B, I would be a fan of a more annoyingly explicit name like `_isConnectedTestOverride` or `_simulatedIsConnectedState`

That would make this easier to understand:
```
if (_isConnected === false || isConnected === false) {
```
Without having to look elsewhere for context.

Reply:
> so we cannot rely on expecting folks to disable their connection to see this component

With a helpful storybook message, I'm not sure I agree with that.  But I feel like we've nit-picked this quite a bit and I think this PR is in great shape as-is.  By all means keep tweaking if desired, but also approving in case you want to merge & iterate in subsequent PRs.

Reply:
> > so we cannot rely on expecting folks to disable their connection to see this component
> 
> With a helpful storybook message, I'm not sure I agree with that. But I feel like we've nit-picked this quite a bit and I think this PR is in great shape as-is. By all means keep tweaking if desired, but also approving in case you want to merge & iterate in subsequent PRs.

All I mean to say here is that this does not work on iOS simulators properly. It will show the message but never hide it. So I take that last statement back, we could instruct them accordingly. I'm going to merge this as-is since it has been growing and growing in history and circle back with some improvements.

Reply:
> If we go for B, I would be a fan of a more annoyingly explicit name like `_isConnectedTestOverride` or `_simulatedIsConnectedState`
> 
> That would make this easier to understand:
> 
> ```
> if (_isConnected === false || isConnected === false) {
> ```
> 
> Without having to look elsewhere for context.

👍 Good advice, thanks. I may eliminate this prop in a follow up PR but if not, I will rename it.